Panic in display logic when interactive mode is (incorrectly?) inferred in CI

A user reported that they are seeing the panic below when they run in CI unless they explicitly pass `--non-interactive`.  We should in this case be detecting we are in a non-interactive environment and disabling the progress display by default - but it appears this detection is not working correctly.

```
panic: fatal: An assertion has failed: action must be non empty
goroutine 32 [running]:
github.com/pulumi/pulumi/pkg/util/contract.failfast(...)
	/home/travis/gopath/src/github.com/pulumi/pulumi/pkg/util/contract/failfast.go:23
github.com/pulumi/pulumi/pkg/util/contract.Assertf(0x0, 0x16d97be, 0x18, 0x0, 0x0, 0x0)
	/home/travis/gopath/src/github.com/pulumi/pulumi/pkg/util/contract/assert.go:33 +0x197
github.com/pulumi/pulumi/pkg/backend/display.makeActionProgress(0x2659b50, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/travis/gopath/src/github.com/pulumi/pulumi/pkg/backend/display/progress.go:60 +0xe8
github.com/pulumi/pulumi/pkg/backend/display.(*ProgressDisplay).refreshColumns(0xc001238000, 0x2659b50, 0x1, 0xc0012cc230, 0x5, 0x5, 0xc0012c2d80, 0x5, 0x5)
	/home/travis/gopath/src/github.com/pulumi/pulumi/pkg/backend/display/progress.go:373 +0x10d
github.com/pulumi/pulumi/pkg/backend/display.(*ProgressDisplay).refreshAllRowsIfInTerminal(0xc001238000)
	/home/travis/gopath/src/github.com/pulumi/pulumi/pkg/backend/display/progress.go:591 +0x36e
github.com/pulumi/pulumi/pkg/backend/display.(*ProgressDisplay).processNormalEvent(0xc001238000, 0x16c72e0, 0xc, 0x1559ec0, 0xc000aba640)
	/home/travis/gopath/src/github.com/pulumi/pulumi/pkg/backend/display/progress.go:966 +0x45a
github.com/pulumi/pulumi/pkg/backend/display.(*ProgressDisplay).processEvents(0xc001238000, 0xc000108050, 0xc000041020)
	/home/travis/gopath/src/github.com/pulumi/pulumi/pkg/backend/display/progress.go:1033 +0x13d
github.com/pulumi/pulumi/pkg/backend/display.ShowProgressEvents.func1(0xc001238000, 0xc000108050, 0xc000041020, 0xc000394000)
	/home/travis/gopath/src/github.com/pulumi/pulumi/pkg/backend/display/progress.go:277 +0x3f
created by github.com/pulumi/pulumi/pkg/backend/display.ShowProgressEvents
	/home/travis/gopath/src/github.com/pulumi/pulumi/pkg/backend/display/progress.go:276 +0x44b
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -20,6 +20,9 @@ CHANGELOG
 - Support exporting older stack versions.
   [#3906](https://github.com/pulumi/pulumi/pull/3906)
 
+- Disable interactive progress display when no terminal size is available.
+  [#3936](https://github.com/pulumi/pulumi/pull/3936)
+
 ## 1.10.1 (2020-02-06)
 - Support stack references in the Go SDK.
   [#3829](https://github.com/pulumi/pulumi/pull/3829)

--- a/pkg/backend/display/progress.go
+++ b/pkg/backend/display/progress.go
@@ -272,10 +272,15 @@ func ShowProgressEvents(op string, action apitype.UpdateKind, stack tokens.QName
 	}
 
 	terminalWidth, terminalHeight, err := terminal.GetSize(int(os.Stdout.Fd()))
-	contract.IgnoreError(err)
-	display.isTerminal = opts.IsInteractive
-	display.terminalWidth = terminalWidth
-	display.terminalHeight = terminalHeight
+	if err == nil {
+		// If the terminal has a size, use it.
+		display.isTerminal = opts.IsInteractive
+		display.terminalWidth = terminalWidth
+		display.terminalHeight = terminalHeight
+	} else {
+		// Else assume we are not displaying in a terminal.
+		display.isTerminal = false
+	}
 
 	go func() {
 		display.processEvents(ticker, events)

